h2o-clustering/src/main/java/water/clustering/api/AssistedClusteringRestApi.java

```diff
@@ -0,0 +1,80 @@
+package water.clustering.api;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.router.RouterNanoHTTPD;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Rest API definition for the assisted clustering function.
+ */
+public class AssistedClusteringRestApi extends RouterNanoHTTPD implements AutoCloseable {
+
+    /**
+     * Default port to bind to / listen on.
+     */
+    private static final int DEFAULT_PORT = 8080;
+    public static final String ASSISTED_CLUSTERING_PORT_KEY = "H2O_ASSISTED_CLUSTERING_API_PORT";
+    private final Consumer<String> flatFileConsumer;
+
+
+    /**
+     * Creates, but not starts assisted clustering REST API. To start the REST API, please use
+     * one of the start methods available.
+     * <p>
+     * The REST API is bound to a default port of 8080, unless specified otherwise by the H2O_ASSISTED_CLUSTERING_API_PORT environment
+     * variable.
+     */
+    public AssistedClusteringRestApi(Consumer<String> flatFileConsumer) {
+        super(getPort());
+        Objects.requireNonNull(flatFileConsumer);
+        this.flatFileConsumer = flatFileConsumer;
+        addMappings();
+    }
+
+    /**
+     * @return Either user-defined port via environment variable or default port to bind the REST API to.
+     */
+    private static int getPort() {
+        final String customPort = System.getenv(ASSISTED_CLUSTERING_PORT_KEY);
+
+        if (customPort == null) {
+            return DEFAULT_PORT;
+        }
+        try {
+            return Integer.parseInt(customPort);
+        } catch (NumberFormatException e) {
+            final String errorMessage = String.format("Unusable port for Assisted clustering REST API to bind to: '%s'", customPort);
+            throw new IllegalArgumentException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public void addMappings() {
+        super.addMappings();
+        addRoute("/clustering/flatfile", AssistedClusteringEndpoint.class, this.flatFileConsumer);
```

Comment:
Just a question to think about (might be irrelevant)... Should this endpoint be secured? Couldn't an enemy call this endpoint with a flat file containing own node?

Reply:
Yes, that could happen.

Reply:
Can be improved in another iteration.